acceptance/tests/pxp-module-puppet/run_puppet_agent_non_root.rb

```diff
@@ -0,0 +1,59 @@
+require 'pxp-agent/test_helper.rb'
+require 'yaml'
+
+test_name  'Run puppet agent as non-root' do
+
+  platform = agents.first.platform
+  skip_test "Test is not compatible with #{platform}" if platform =~ /windows/
+
+  step 'create non-root user on all nodes' do
+    on(agents, puppet("resource user foo ensure=present managehome=true"))
+
+    #TODO fix this only one agent ???
+    @user_home_dir = YAML.load(on(agents.first, puppet("resource user foo --to_yaml")).stdout)['user']['foo']['home']
+    @user_puppetlabs_dir = "#{@user_home_dir}/.puppetlabs"
+    @user_puppet_dir = "#{@user_puppetlabs_dir}/etc/puppet"
+    @user_pxp_module_dir = "#{@user_puppetlabs_dir}/etc/pxp-agent/modules"
+    on(agents, "mkdir -p #{@user_puppet_dir}")
+    on(agents, "mkdir -p #{@user_pxp_module_dir}")
+  end
+
+  teardown do
+    on(agents, "kill -9 `pidof pxp-agent`", :accept_all_exit_codes => true)
+    on(agents, puppet("resource user foo ensure=absent managehome=true"))
+  end
+
+  step 'Ensure pxp-agent is stopped' do
+    on(agents, puppet("resource service pxp-agent ensure=stopped enable=false"))
+  end
+
+  step 'Copy certs and keys to new users home directory' do
+    agents.each do |agent|
+      puppet_ssldir = on(agent, puppet('config print ssldir')).stdout.chomp
+      #ssl-key ssl-ca-cert ssl-cert
+      on(agent, "cp -R #{puppet_ssldir} #{@user_puppet_dir}")
+    end
+  end
+
+  step 'Attempt start of pxp-agent as non-root user' do
+    on(agents, puppet("resource service puppet ensure=running enable=true"))
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+
+      ssl_config = {
+          :ssl_key        => "#{@user_puppet_dir}/ssl/private_keys/#{agent}.pem",
+          :ssl_ca_cert    => "#{@user_puppet_dir}/ssl/certs/ca.pem",
+          :ssl_cert       => "#{@user_puppet_dir}/ssl/certs/#{agent}.pem"
+      }
+
+      create_remote_file(agent, "#{@user_pxp_module_dir}/pxp-agent.conf", pxp_config_json(master, agent, ssl_config).to_s)
```

Comment:
Config should go in `~/.puppetlabs/etc/pxp-agent/pxp-agent.conf`. Right now you're instead putting it in the modules subdirectory.